lib/Backend/GlobOptArrays.cpp

```diff
@@ -1829,10 +1886,17 @@ void GlobOpt::ArraySrcOpt::Optimize()
                 eliminatedUpperBoundCheck,
                 instr->m_func);
 
-        if (baseOwnerInstr)
+        if (baseOwnerInstr != nullptr)
         {
-            Assert(baseOwnerInstr->GetSrc1() == baseOpnd);
-            baseOwnerInstr->ReplaceSrc1(baseArrayOpnd);
+            if (baseOwnerInstr->GetSrc1() == baseOpnd)
+            {
+                baseOwnerInstr->ReplaceSrc1(baseArrayOpnd);
+            }
+            else
+            {
+                Assert(baseOwnerInstr->GetSrc2() == baseOpnd);
+                baseOwnerInstr->ReplaceSrc1(baseArrayOpnd);
```

Comment:
>ReplaceSrc1 [](start = 32, length = 11)

this should be ReplaceSrc2(), no?

Reply:
Yeah, that was supposed to be ReplaceSrc2.